File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Lab/03_Combinations.js
function combinations(input) {
    let targetNumber = Number(input[0]);

    let validCombinationsCount = 0;

    for (let x1 = 0; x1 <= targetNumber; x1++) {
        for (let x2 = 0; x2 <= targetNumber; x2++) {
            for (let x3 = 0; x3 <= targetNumber; x3++) {
                let currentSum = x1 + x2 + x3;
                if (currentSum == targetNumber) {
                    validCombinationsCount++;
                }
            }
        }
    }

    console.log(validCombinationsCount);
}


// combinations(["25"]);

File: Python_Courses/Python_Basics/Preparation/06_Nested_Loops_Lab/03_Combinations.py
number = int(input())

combinations = 0
for n1 in range(number + 1):
    for n2 in range(number + 1):
        for n3 in range(number + 1):
            if n1 + n2 + n3 == number:
                combinations += 1

print(combinations)
